### DISCUSSION TITLE
i18n for D3 documentation

### DISCUSSION THREAD

**Discussion by Benbinbin** [d3/d3#3693] [Q&A]
It's satisfied to know the update of [D3 documentation](https://d3js.org/), I think it will be more helpful to provide other language version, I'd like to contribute to translate the documentation, any plan about it?

**Comment by Fil** [comment]
Hi, thanks for asking the question! Here are a few thoughts.

Everyone is free, of course, to translate parts of the documentation as they see fit. Translating is a great way to consolidate your knowledge of something and discover new details, and sharing a translation can help more users, especially when they want to onboard.

However, I don't think we'll ever start a project to have an “official” complete multilingual and “manually translated” documentation. For a start, we certainly don't have the bandwidth to organize this. Even just coordinating community-maintained translations seems like a huge effort. Plus, the documentation is a large living document, and having to reflect each change into translations would feel like a burden.

If you translate some of these pages, I would recommend that you focus on the “introduction” part of the modules you're interested in—maybe because you want to learn more about them, or to teach them. For the detailed descriptions of every method in the API, though, it might be better to not translate, but instead give enough background so that readers can read (or sometimes decipher) the machine translated version.

On this topic, the machine translation experience (with google translate) is not great at the moment on this website, and this is a domain where might be possible to improve. One idea would be to experiment with the [translate=no](https://html.spec.whatwg.org/multipage/dom.html#the-translate-attribute) attribute on all the symbols, so that for example d3.color doesn't get translated (in French) to “d3.couleur”.

**Reply by mbostock** [reply]
react.dev gives a good indication of how this might work (if we had the resources to support it):

https://github.com/reactjs/translations.react.dev

I think in essence we would create separate repos for each language under the d3 organization, host them on {lang}.d3js.org, and have the same automation to deploy the site via GitHub Actions and Pages. It’s a big effort though!

**Reply by Benbinbin** [reply]
@Fil thanks for your reply
